pkg/hub/manager.go

```diff
@@ -79,22 +90,31 @@ func RunControllerManager(ctx context.Context, controllerContext *controllercmd.
 		clusterClient,
 		workClient,
 		configClient,
+		addOnClient,
 		clusterInformers.Cluster().V1().ManagedClusters(),
 		clusterInformers.Cluster().V1alpha1().ManagedClusterSets(),
 		workInformers.Work().V1().ManifestWorks(),
 		configInformers.Submarineraddon().V1alpha1().SubmarinerConfigs(),
+		addOnInformers.Addon().V1alpha1().ManagedClusterAddOns(),
 		controllerContext.EventRecorder,
 	)
-
 	go clusterInformers.Start(ctx.Done())
 	go workInformers.Start(ctx.Done())
 	go kubeInformers.Start(ctx.Done())
 	go configInformers.Start(ctx.Done())
 	go apiExtensionsInformers.Start(ctx.Done())
+	go addOnInformers.Start(ctx.Done())
 	go submarinerBrokerCRDsController.Run(ctx, 1)
 	go submarinerBrokerController.Run(ctx, 1)
 	go submarinerAgentController.Run(ctx, 1)
 
+	mgr, err := addonmanager.New(controllerContext.KubeConfig)
+	if err != nil {
+		return err
+	}
+	mgr.AddAgent(submarineraddonagent.NewAddOnAgent(rand.String(5), kubeClient, controllerContext.EventRecorder))
```

Comment:
using rand here might be a problem. It will change after each restart

Reply:
instead of using a fixed agent name